src/target/xtensa/xtensa.c

```diff
@@ -19,307 +20,269 @@
-	{ "mesave", 0x6C, XT_REG_SPECIAL, 0 },
-	{ "mesr", 0x6D, XT_REG_SPECIAL, 0 },
-	{ "mecr", 0x6E, XT_REG_SPECIAL, 0 },
-	{ "mevaddr", 0x6F, XT_REG_SPECIAL, 0 },
-	{ "a0", XT_REG_IDX_AR0, XT_REG_RELGEN, 0 },	/* WARNING: For these registers, regnum points to the */
-	{ "a1", XT_REG_IDX_AR1, XT_REG_RELGEN, 0 },	/* index of the corresponding ARxregisters, NOT to */
-	{ "a2", XT_REG_IDX_AR2, XT_REG_RELGEN, 0 },	/* the processor register number! */
-	{ "a3", XT_REG_IDX_AR3, XT_REG_RELGEN, 0 },
-	{ "a4", XT_REG_IDX_AR4, XT_REG_RELGEN, 0 },
-	{ "a5", XT_REG_IDX_AR5, XT_REG_RELGEN, 0 },
-	{ "a6", XT_REG_IDX_AR6, XT_REG_RELGEN, 0 },
-	{ "a7", XT_REG_IDX_AR7, XT_REG_RELGEN, 0 },
-	{ "a8", XT_REG_IDX_AR8, XT_REG_RELGEN, 0 },
-	{ "a9", XT_REG_IDX_AR9, XT_REG_RELGEN, 0 },
-	{ "a10", XT_REG_IDX_AR10, XT_REG_RELGEN, 0 },
-	{ "a11", XT_REG_IDX_AR11, XT_REG_RELGEN, 0 },
-	{ "a12", XT_REG_IDX_AR12, XT_REG_RELGEN, 0 },
-	{ "a13", XT_REG_IDX_AR13, XT_REG_RELGEN, 0 },
-	{ "a14", XT_REG_IDX_AR14, XT_REG_RELGEN, 0 },
-	{ "a15", XT_REG_IDX_AR15, XT_REG_RELGEN, 0 },
-
-	{ "pwrctl", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pwrstat", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "eristat", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_itctrl", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_claimset", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_claimclr", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_lockaccess", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_lockstatus", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "cs_authstatus", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "fault_info", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_id", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_ctrl", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_stat", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_data", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_addr", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_pctrigger", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_pcmatch", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_delay", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_memstart", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "trax_memend", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmg", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmoc", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pm0", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pm1", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmctrl0", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmctrl1", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmstat0", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "pmstat1", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "ocd_id", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "ocd_dcrclr", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "ocd_dcrset", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "ocd_dsr", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
-	{ "ddr", 0x68, XT_REG_DEBUG, XT_REGF_NOREAD },
+#define XT_PS_REG_NUM               (0xe6U)
+#define XT_EPS_REG_NUM_BASE         (0xc0U)	/* (EPS2 - 2), for adding DBGLEVEL */
```

Comment:
@ianstcdns I changed the macro names. Looks more correct to me.  Same as `XT_EPC_REG_NUM_BASE`

Reply:
sure.  there's no PC register so EPC base is more accurate.

Reply:
Created https://review.openocd.org/c/openocd/+/7227